### ISSUE TITLE
node taint failed

### ISSUE THREAD

**Issue by ArchiFleKs** [coreos/kube-aws#232]
Hi, I'm not really sure I fully understand how taint works. But I noticed that on v0.9 controllers node are not marked as unschedulable, nor in the apiserver parameters.

In the userdata, there is a command that failed : 
https://github.com/coreos/kube-aws/blob/master/config/templates/cloud-config-controller#L479

```
+ echo tainting this node
24178] hyperkube[5]: tainting this node
24404] hyperkube[5]: + hostname=ip-10-0-2-7.eu-west-1.compute.internal
24615] hyperkube[5]: + kubectl='/kubectl --server=http://127.0.0.1:8080'
24906] hyperkube[5]: + taint='/kubectl --server=http://127.0.0.1:8080 taint node ip-10-0-2-7.eu-west-1.compute.internal'
25187] hyperkube[5]: + /kubectl --server=http://127.0.0.1:8080 taint node ip-10-0-2-7.eu-west-1.compute.internal node.alpha.kubernetes.io/role:master:NoSchedule
57690] hyperkube[5]: error: at least one taint update is required
59076] hyperkube[5]: + echo done.
59381] hyperkube[5]: done.
59673] hyperkube[5]: + echo uncordoning this node
59947] hyperkube[5]: uncordoning this node
60249] hyperkube[5]: + /kubectl --server=http://127.0.0.1:8080 uncordon ip-10-0-2-7.eu-west-1.compute.internal
33116] hyperkube[5]: node "ip-10-0-2-7.eu-west-1.compute.internal" already uncordoned
36197] hyperkube[5]: + echo done.
36576] hyperkube[5]: done.
```

From what I understand, taint should be a K=V type like in this issue : https://github.com/kubernetes/kubernetes/issues/33530 
But apparently this not right either (https://github.com/kubernetes/kubernetes/issues/36272).

Again from what I understand there should be : 
- a label : node.alpha.kubernetes.io/role=master
- a taint : node.alpha.kubernetes.io/role=master and not dedicated (that's the case in master branch)

Is that correct and if that's the case it just a missing "=" instead of ":"  ?

**Comment by whereisaaron** [comment]
Hi @ArchiFleKs! You are correct it should be '=' and this has been patched by PR #199, combined with another fix for the taint-and-cordon task on controllers (Ref #215 #199 for controller, #231 for similar fix on workers).

**Comment by mumoshu** [comment]
@ArchiFleKs Thanks for your report 👍
As @whereisaaron explained, this should be fixed in v0.9.3-rc.3 released today. Would you mind trying it and reopen if the issue still persists?
